app.js

```diff
@@ -11,62 +12,100 @@ const teamTwoSubtractButton = document.getElementById('team-two-subtract-button'
+
+
 
 // create an array to hold on to the state of past games
 
 let name1 = '';
-let name2 =  '';
+let name2 = '';
 let score1 = 0;
 let score2 = 0;
+let pastGames = [];
 
 nameFormButton.addEventListener('click', () => {
     // get the name data from the form
+    const teamOneName = teamOneInput.value;
+    teamOneLabel.textContent = teamOneName;
+    const teamTwoName = teamTwoInput.value;
+    teamTwoLabel.textContent = teamTwoName;
+    
 
     // set the state to this data from the form
-
+    name1 = teamOneInput.value;
+    name2 = teamTwoInput.value;
+    
+    
     // reset the form values
+    
+    teamOneInput.value = '';
+    teamTwoInput.value = '';
+    
+    
+    refreshCurrentGameEl();
 
-    // refresh the current game element with new data by calling the appropriate function
 });
 
 
 teamOneAddButton.addEventListener('click', () => {
-    // increment the current state for team one's score
     
-    // refresh the current game element with new data by calling the appropriate function
+    currentGameEl.textContent = '';
+    score1++;
+    
+    refreshCurrentGameEl();
+
+
 });
 
 teamTwoAddButton.addEventListener('click', () => {
-    // increment the current state for team two's score
+    
+    currentGameEl.textContent = '';
+    score2++;
 
-    // refresh the current game element with new data by calling the appropriate function
+    refreshCurrentGameEl();
 });
 
 teamOneSubtractButton.addEventListener('click', () => {
-    // decrement the current state for team one's score
+    
+    currentGameEl.textContent = '';
+    score1--;
 
-    // refresh the current game element with new data by calling the appropriate function
+    refreshCurrentGameEl();
 });
 
 teamTwoSubtractButton.addEventListener('click', () => {
-    // decrement the current state for team two's score
-
-    // refresh the current game element with new data by calling the appropriate function
+    
+    currentGameEl.textContent = '';
+    score2--;
+    
+    refreshCurrentGameEl();
 });
 
 finishGameButton.addEventListener('click', () => {
     
-    // add the current game to an array of games in state
-    // it will be helpful to keep track of these games as objects with 4 properties, one for each piece of state we're tracking
-    // for example, make an object like this: { name1: 'ducks', name2: 'bears' ,score1: 1, score2: 2 } 
-    // then push it to your array in state
-    // (be sure to make a new object. do not declare the object in global scope and mutate it for reuse. This would cause difficult bugs)
     
-    displayAllGames();
+    // currentGameEl.push(pastGames);
 
-    // reset the state to zero and empty strings
-    
-    // refresh the current game element with new data by calling the appropriate function
+    const currentGameData = {
+        name1: name1,
+        name2: name2,
+        score1: score1,
+        score2: score2, 
+    };
+
+
+    pastGames.push(currentGameData);
+    pastGamesEl.textContent = '';
```

Comment:
This (line 100) is already happening in displayAllGames() below so is unnecessary here.